src/NServiceBus.Core/Notifications/EventAggregator.cs

```diff
@@ -0,0 +1,20 @@
+namespace NServiceBus
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    class EventAggregator
+    {
+        public EventAggregator(NotificationSubscriptions subscriptions)
+        {
+            this.subscriptions = subscriptions;
+        }
+
+        public Task Raise<T>(T @event) where T : INotification
+        {
+            return Task.WhenAll(subscriptions.Get<T>().Select(s => s.Invoke(@event)));
```

Comment:
> Do we wanna abort message processing when we can't write a performance counter or something like that?

That's how the current code behave: https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Performance/Statistics/ReceivePerformanceDiagnosticsBehavior.cs#L31